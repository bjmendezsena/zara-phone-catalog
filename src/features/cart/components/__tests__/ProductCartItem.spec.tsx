import {
  CartItem,
  ProductCartItem,
  ProductCartItemProps,
} from "@/features/cart";
import { render } from "@/tests";

describe(`<${ProductCartItem.name} />`, () => {
  const onRemoveMock = jest.fn();
  const item: CartItem = {
    productId: "1",
    color: "Black",
    storage: "128GB",
    brand: "BrandX",
    name: "Product Name",
    price: 999.99,
    imageUrl: "/path/to/image.jpg",
    quantity: 2,
  };

  const getComponent = (props: Partial<ProductCartItemProps> = {}) => {
    return render(
      <ProductCartItem item={{ ...item, ...props }} onRemove={onRemoveMock} />
    );
  };
  it("renders product details correctly", async () => {
    const { asFragment } = getComponent();

    expect(asFragment()).toMatchSnapshot();
  });
  it("calls onRemove when remove button is clicked", async () => {
    const { getByTestId } = getComponent();

    const removeButton = getByTestId("remove-cart-item");
    removeButton.click();

    expect(onRemoveMock).toHaveBeenCalledWith(item.productId);
  });
});
