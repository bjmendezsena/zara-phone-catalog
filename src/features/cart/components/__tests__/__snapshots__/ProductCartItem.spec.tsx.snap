// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`<ProductCartItem /> renders product details correctly 1`] = `
<DocumentFragment>
  <div
    class="flex w-full h-[364px]"
  >
    <div
      class="flex space-x-8"
    >
      <div
        class="h-full w-[262px] p-2 flex items-center justify-center rounded-md overflow-hidden"
      >
        <img
          alt="BrandX Product Name"
          class="w-full h-[90%] object-cover"
          data-priority="true"
          height="100"
          src="/path/to/image.jpg"
          width="100"
        />
      </div>
      <div
        class="flex flex-col gap-4 pt-[20%] py-[65px] h-full"
      >
        <div
          class="flex flex-col"
        >
          <span>
            BrandX Product Name
          </span>
          <span>
            128GB | Black
          </span>
        </div>
        <span>
          999.99 EUR
        </span>
        <span
          aria-label="Remove Product Name from cart"
          class="text-red-500 hover:text-red-700 cursor-pointer transition-colors duration-200"
          data-testid="remove-cart-item"
        >
          Eliminar
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
